### PR TITLE
issue #183: analyse route using fastapi

### PR DESCRIPTION
**Pull request by k-allagbe**
- [x] route
- [x] tests

### CODE REVIEW

File: app/controllers/data_extraction.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import os
+from typing import BinaryIO
+
+from pipeline import LabelStorage, analyze
+
+from app.constants import UPLOAD_FOLDER
+
+
+def extract_data(files: dict[str, BinaryIO], ocr, gpt):
```
Comment (ChromaticPanic) [inline app/controllers/data_extraction.py:9]: Return type?

File: app/dependencies.py [hunk 2]
```diff
@@ -6,3 +6,17 @@ def get_connection_manager(request: Request):
     Returns the app's ConnectionManager instance.
     """
     return request.app.connection_manager
+
+
+def get_ocr(request: Request):
```
Comment (ChromaticPanic) [inline app/dependencies.py:11]: Return type?

File: app/sanitization.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import re
+
+
+def custom_secure_filename(filename):
```
Comment (ChromaticPanic) [inline app/sanitization.py:4]: need types

File: tests/test_app.py [hunk 4]
```diff
@@ -52,3 +54,43 @@ def test_rollbacks(self):
...
+        file_content_1 = b"Sample content 1"
+        file_content_2 = b"Sample content 2"
+
+        files = [
+            ("files", ("file1.txt", file_content_1, "text/plain")),
+            ("files", ("file2.txt", file_content_2, "text/plain")),
+        ]
+
+        with TestClient(app) as client:
+            response = client.post("/analyze", files=files)
```
Comment (ChromaticPanic) [inline tests/test_app.py:78]: Need more test cases?
Empty file
file list with one or many empty files
empty file list

File: tests/test_sanitization.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        self.assertEqual(custom_secure_filename("project#1!.zip"), "project1.zip")
+
+    def test_complex_filename(self):
+        self.assertEqual(custom_secure_filename("my<>file|name?.exe"), "myfilename.exe")
+
+    def test_no_changes_needed(self):
+        self.assertEqual(
+            custom_secure_filename("valid_filename.doc"), "valid_filename.doc"
+        )
+
```
Comment (ChromaticPanic) [inline tests/test_sanitization.py:29]: sql injection test case
5; DROP TABLE users; --
DROP Table users;--

File: app/controllers/data_extraction.py (hunk 1 above)
Comment (k-allagbe) [reply app/controllers/data_extraction.py]: done. Although this one can be inferred by python from the `analyze` function.

File: app/dependencies.py (hunk 2 above)
Comment (k-allagbe) [reply app/dependencies.py]: done

File: app/sanitization.py (hunk 3 above)
Comment (k-allagbe) [reply app/sanitization.py]: done

File: tests/test_app.py (hunk 4 above)
Comment (k-allagbe) [reply tests/test_app.py:78]: done. I now raise 422 in those cases in the route.

File: tests/test_sanitization.py (hunk 5 above)
Comment (k-allagbe) [reply tests/test_sanitization.py:29]: Done. Although, note that that function is temporary. We'll be moving away from using the filesystem. Check ai-cfia/fertiscan-pipeline#66.

**Review by ChromaticPanic — APPROVED** [review]

**Pull request merged**